fyrox-ui/src/ttf.rs

```diff
@@ -294,3 +295,62 @@ impl Font {
         }
     }
 }
+
+/// Font builder allows you to load fonts in declarative manner.
+pub struct FontBuilder<'a> {
+    height: Option<f32>,
+    char_set: Option<Cow<'a, [Range<u32>]>>,
+}
+impl<'a> FontBuilder<'a> {
+    const DEFAULT_HEIGHT: f32 = 16.0;
+
+    /// Creates a default FontBuilder.
+    pub fn new() -> Self {
+        Self {
+            height: None,
+            char_set: None,
+        }
+    }
+
+    /// Sets the desired font height for the produced font.
+    #[inline]
+    pub fn with_height(mut self, height: f32) -> Self {
+        self.height = Some(height);
+        self
+    }
+
+    /// Sets the desired character set for the produced font.
+    #[inline]
+    pub fn with_char_set(mut self, char_set: impl Into<Cow<'a, [Range<u32>]>>) -> Self {
+        self.char_set = Some(char_set.into());
+        self
+    }
+
+    /// Creates a new font from the data at the specified path.
+    pub async fn build_from_file(self, path: impl AsRef<Path>) -> Result<Font, &'static str> {
```

Comment:
I decided to make multiple build methods because of the mix of `async` and sync building.

You could also do something like this:
```rust
pub enum FontSource<'a> {
    BuiltIn,
    Path(&'a Path),
    Memory(&'a [u8]),
}

pub fn with_source(self, source: FontSource) -> Self { ... }

pub fn build(self) -> ... {}
```

Or something like this ⬇️ - but I think it's better if the builder is lazy:
```rust
pub async fn with_data_from_file(self, path: impl AsRef<Path>) -> Result<Self, &'static str> { ... }
```

A more complicated way to do it, but with perhaps a more familiar API is:
```rust
pub struct BuiltInFontSource;
pub struct FontBuilder<'a, Source = BuiltInFontSource> { source: Source, ... }
impl<'a> FontBuilder<'a, BuiltInFontSource> {
    pub fn with_data_from_memory(self, data: impl Deref<Target = [u8]>) -> FontBuilder<'a, &'a [u8]> { ... }
    pub fn with_data_from_file(self, path: impl AsRef<Path>) -> FontBuilder<'a, &'a Path> { ... }
    pub fn build(self) -> Result<Font, &'static str> { ... }
}
impl<'a> FontBuilder<'a, &'a [u8]> {
    pub fn build(self) -> Result<Font, &'static str> { ... }
}
impl<'a> FontBuilder<'a, &'a Path> {
    pub async fn build(self) -> Result<Font, &'static str> { ... }
}
```

Reply:
Current solution is perfectly fine, thanks!

Reply:
Thanks for the review!